CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+# How To Contribute To Rototiller
+
+## Getting Started
+
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Fork the [Rototiller repository on GitHub](https://github.com/puppetlabs/rototiller)
+
+## Making Changes
+
+* Create a topic branch from where you want to base your work.
+  * This is the `master` branch in the case of rototiller
+  * To quickly create a topic branch based on master use `git checkout -b my_contribution master`. Do not work directly on the `master` branch.
+* Make commits of logical _working_ and _functional_ units.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure your commit messages are in the proper format.
+
+        (BKR-1234) Make the example in CONTRIBUTING imperative and concrete
+
+        Without this patch applied the example commit message in the CONTRIBUTING
+        document is not a concrete example.  This is a problem because the
+        contributor is left to imagine what the commit message should look like
+        based on a description rather than an example.  This patch fixes the
+        problem by making the example concrete and imperative.
+
+        The first line is a real life imperative statement with a ticket number
+        from our issue tracker.  The body describes the behavior without the patch,
+        why this is a problem, and how the patch fixes the problem when applied.
+
+* Make sure you have added [RSpec](http://rspec.info/) tests that exercise your new code.  These test should be located in the appropriate `rototiller/spec/` subdirectory.  The addition of new methods/classes or the addition of code paths to existing methods/classes requires additional RSpec coverage.
+  * One should **NOT USE** the deprecated `should`/`stub` methods - **USE** `expect`/`allow`. Use of deprecated RSpec methods will result in your patch being rejected.  See a nice blog post from 2013 on [RSpec's new message expectation syntax](http://teaisaweso.me/blog/2013/05/27/rspecs-new-message-expectation-syntax/).
+* Run the spec unit tests to assure nothing else was accidentally broken, using `rake test`
+  * **Bonus**: if possible ensure that `rake test` runs without failures for additional Ruby versions (1.9, 2.0, etc). Rototiller supports Ruby 1.9+, and breakage of support for other rubies will cause a patch to be rejected.
```

Comment:
There has been discussion about beaker ditching support for < 2.1 which would be nice.  But we probably need to keep this here until that becomes official.